Update to 1.20.6

thelooter opened the pull request:
Update to 1.20.6

thelooter committed:
chore(update): Update to 1.20.6

thelooter committed:
chore(update): Uncomment previously commented code

Insprill commented on build.gradle.kts:
```diff
@@ -17,6 +17,9 @@ repositories {
 	maven("https://hub.spigotmc.org/nexus/content/repositories/public/")
 	maven("https://repo.md-5.net/content/groups/public/")
 	maven("https://repo.papermc.io/repository/maven-public/")
+	maven(url = "https://s01.oss.sonatype.org/content/repositories/snapshots/") {
+		name = "sonatype-oss-snapshots"
+	}
```
```suggestion
maven("https://s01.oss.sonatype.org/content/repositories/snapshots/")

original-codematrix replied:
Should not be needed anymore in the newest version due to kyori adventure-bom has been released as 2.17.0

Insprill commented on src/main/java/be/seeseemelk/mockbukkit/block/state/EnderChestMock.java:
```diff
@@ -75,4 +76,11 @@ public boolean isOpen()
 		return isOpen;
 	}
 
+	@Override
+	public boolean isBlocked()
+	{
+		// TODO Auto-generated method stub
+		throw new UnimplementedOperationException();
+	}
+
```
Does Paper have Chest and EnterChest extend from a common class to implement this?

Thorinwasher replied:
No, the chest implementation is different enough due to double chests.

Insprill commented on src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java:
```diff
@@ -887,6 +887,7 @@ public void setFallDistance(float distance)
 	}
 
 	@Override
+	@Deprecated(forRemoval = true,since = "1.20.6")
```
```suggestion
	@Deprecated(forRemoval = true, since = "1.20.6")

Insprill commented on src/main/java/be/seeseemelk/mockbukkit/entity/LivingEntityMock.java:
```diff
@@ -1283,4 +1283,6 @@ public float getSidewaysMovement()
 	}
 
 
+
+
```
```suggestion

Insprill reviewed (changes requested):
Some preliminary nitpicks

thelooter committed:
chore(update): Update upstream Paper dependency

Thorinwasher committed:
Implement version info service

Thorinwasher committed:
Create registry access mock

Thorinwasher committed:
Update generated resources

Thorinwasher committed:
Fix compile issues, add a test

Thorinwasher committed:
Add new keyed objects

Thorinwasher committed:
Registries now rely on registry keyes instead of class types

Thorinwasher committed:
Remove unsupported information on enchantment

Thorinwasher committed:
Fix all tests not failing becase of UnimplementedOperationException

Note that I had to remove some tests here related to EnchantmentMock, as some methods in that has been removed

Thorinwasher committed:
Add more data to ItemType and BlockType

Thorinwasher committed:
Fix invalid tests

Thorinwasher committed:
Add translation keys to item type and block type

Thorinwasher committed:
Fix all remaining UnimplementedOperationExceptions for 1.20.6

Thorinwasher committed:
Fix floating point precision errors from prev commit

Thorinwasher committed:
Javadoc fix

Thorinwasher committed:
Style correction

Thorinwasher committed:
Code cleanup

Thorinwasher committed:
Fix some code smells

Thorinwasher committed:
Update RegistryAccessMock.java

Thorinwasher committed:
Avoid comparing class names, compare classes instead

GrayR0ot commented:
Are changes done ?

Insprill commented:
Please do not create unnecessary noise on PRs. It's not going to get it merged faster.

thelooter committed:
Merge branch 'v1.20' into feat/1.20.6

GrayR0ot commented:
Created https://github.com/MockBukkit/MockBukkit/pull/1015 to fix build issues, they came from 1.20 rebase

GrayR0ot committed:
fix: Fixed getLootTable, isCollidable, hasCollision, getAnvilCost, setLootTable, getDefaultAttributeModifiers, translationKey methods (Not Implemented) (#1015)

Thorinwasher committed:
Merge remote-tracking branch 'upstream/v1.20' into feat/1.20.6

# Conflicts:
#	src/main/java/be/seeseemelk/mockbukkit/inventory/meta/BannerMetaMock.java
#	src/main/java/be/seeseemelk/mockbukkit/inventory/meta/ItemMetaMock.java

Thorinwasher committed:
Fix build issues

Thorinwasher committed:
Fix weird ?encoding? issue

Thorinwasher committed:
Fix test failures

Thorinwasher committed:
Fix registry mock duplicate keyed object creation due to class loading

Thorinwasher committed:
Stub method

Insprill committed:
Code style

Insprill commented on src/main/java/be/seeseemelk/mockbukkit/services/ServerBuildInfoMock.java:
```diff
@@ -0,0 +1,133 @@
+	private final String minecraftVersion;
+	private static final String BUILD_DEV = "DEV";
+
+	public ServerBuildInfoMock()
+	{
+		try (InputStream inputStream = MockBukkit.class.getResourceAsStream("/version_info.properties"))
+		{
+			Properties properties = new Properties();
+			properties.load(inputStream);
+			this.minecraftVersion = properties.getProperty("minecraft.version");
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+	}
```
We already have a system for this that doesn't require all this overhead. Look at `BuildParameters.java.peb` and the `blossom` block in the `build.gradle.kts`.

Insprill commented on src/main/java/be/seeseemelk/mockbukkit/services/ServerBuildInfoMock.java:
```diff
@@ -0,0 +1,133 @@
+	public @NotNull String minecraftVersionName()
+	{
+		return this.minecraftVersion;
+	}
+
+	@Override
+	public @NotNull OptionalInt buildNumber()
+	{
+		return OptionalInt.empty();
+	}
+
+	@Override
+	public @NotNull Instant buildTime()
+	{
+		return Instant.now();
+	}
+
+	@Override
+	public @NotNull Optional<String> gitBranch()
+	{
+		return Optional.empty();
+	}
+
+	@Override
+	public @NotNull Optional<String> gitCommit()
+	{
+		return Optional.empty();
+	}
```
If we're not going to implement these, we shouldn't implement them. We shouldn't be returning defaults.

On that note, this would also be extremely trivial to implement with the BuildParameters template class. We already get some git info in the build file for the project version outside of actions.

Thorinwasher replied:
Throwing UnimplementedOperationException is not an option as most tests in mockbukkit would be ignored.

I don't understand how the mockbukkit actions system works. But how would we get the branch and commit and buildnumber when building a new instance?

Do we need to modify actions for this?

Insprill replied:
All you'd have to do is add new fields to Build Parameters, and set then in the blossom block in the build.gradle.kts. We already have a method or two in there was calls git, so you'd just need to add/modify them to get the info you need.

Thorinwasher replied:
Ah, using git from gradle is quite obvious when I think about it. Anyhow, what about the buildNumber method? What would be the mockbukkit equivalent of that; as I said before, we can nott throw unimplementedoperationexception

Insprill replied:
We can get the build number from GitHub actions, IIRC it's the `GITHUB_RUN_NUMBER` we'd want (all variables can be found [here](https://docs.github.com/en/actions/learn-github-actions/variables)). If it's a local build then we can just return empty.

Insprill commented on src/main/java/be/seeseemelk/mockbukkit/services/ServerBuildInfoMock.java:
This class needs more proper tests, not just "does it return not null".

Thorinwasher replied:
Most of the properties of that class are ever changing, it therefore becomes hard to test without having to reimplement the code your testing in the test to see if it works (which makes the test invalid).

I believe this would be the case for the following methods:
- minecraftVersionId
- minecraftVersionName
- buildNumber
- buildTime
- ?gitBranch?
- gitCommit

These properties either change with every new update, or would be dependant on the branch your using, which should not be in the test, as that's also ever changing.

So the test has to be dynamic, but also actually test stuff.

Insprill replied:
I was thinking more along the lines of using regex patterns to ensure the returned values are sane, I believe we do this in the UnsafeMockTest. Should've been more clear, sorry.

Thorinwasher replied:
Okay, that makes more sense, I will look into that.

Insprill commented on build.gradle.kts:
```diff
@@ -91,6 +91,13 @@ tasks {
 	jacoco {
 		toolVersion = "0.8.12"
 	}
+
+	processResources {
+		filesMatching("**/version_info.properties") { expand(mapOf(
+			"minecraft_version" to project.properties["paper.api.full-version"].toString().split("-")[0],
+			"paper_api_version" to project.properties["paper.api.full-version"]
+		))}
+	}
```
See other comment, remove this.

Insprill committed:
Remove duplicate method in EnchantmentMock

Insprill committed:
Suppress NonExtendableApiUsage in ServerBuildInfoMock

Insprill reviewed (changes requested):
Looks pretty good, just some concerns with the `ServerBuildInfoMock`.

Thorinwasher committed:
Use buildparameters to determine minecraft version

Thorinwasher committed:
Add build time into build parameters

Thorinwasher committed:
Add test to asString method

Will require modification whenever other params get implemented

Thorinwasher committed:
Implement git branch and commit build information

Thorinwasher committed:
Use regex on tests

Thorinwasher committed:
Syntax error + test failures fix

Thorinwasher committed:
Add build number to server build info

Insprill reviewed (approved):
LGTM!

thelooter committed:
Merge branch 'v1.20' into feat/1.20.6
